-Wpessimizing-move

### Bug Description

Another warning from gcc
```
/home/thu/projects/moose/framework/build/header_symlinks/SharedPool.h:85:91: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
   85 |                                ExternalDeleter{std::weak_ptr<SharedPool<T> *>{_this_ptr}}));
      |                                                                                           ^
/home/thu/projects/moose/framework/build/header_symlinks/SharedPool.h:85:91: note: remove 'std::move' call
```

### Steps to Reproduce

Compile in opt with g++>=13

### Impact

Warnings during compilation.

### [Optional] Diagnostics

_No response_

--- a/framework/include/utils/SharedPool.h
+++ b/framework/include/utils/SharedPool.h
@@ -81,8 +81,8 @@ public:
     if (_pool.empty())
     {
       _num_created++;
-      return std::move(PtrType(new T(std::forward<Args>(args)...),
-                               ExternalDeleter{std::weak_ptr<SharedPool<T> *>{_this_ptr}}));
+      return PtrType(new T(std::forward<Args>(args)...),
+                     ExternalDeleter{std::weak_ptr<SharedPool<T> *>{_this_ptr}});
     }
     else
     {

